static/app/icons/svgIcon.tsx

```diff
@@ -3,7 +3,7 @@ import {withTheme} from '@emotion/react';
 
 import {Aliases, Color, IconSize, Theme} from 'app/utils/theme';
 
-type Props = React.SVGAttributes<SVGElement> & {
```

Comment:
This is what's happening underneath:

```
interface SVGElement extends Element { }
interface SVGSVGElement extends SVGElement { }
```

So weird, but there's probably a reason for it? I'm glad someone is writing boilerplate so we don't have to.

Reply:
well our typing here was just wrong. We're spreading into a `svg` element, but had it typed as the "generic" catchall `SVGElement`, not the element specific `SVGSVGElement`. For example, `SVGPathElement` also exists for `<path>`

Reply:
The problem with react 17 is that the keyboard event types are now a bit more specific, and only the more concrete dom elements will have the correct keyboard types, so typing as a generic SVGElement results in incompatible onKeyDown types.